frontend/src/_helpers/appUtils.js

```diff
@@ -398,26 +398,26 @@ export const executeAction = (_ref, event, mode, customVariables) => {
       case 'set-custom-variable': {
         const key = resolveReferences(event.key, _ref.state.currentState, undefined, customVariables);
         const value = resolveReferences(event.value, _ref.state.currentState, undefined, customVariables);
-        const customAppVariables = { ..._ref.state.currentState.variables };
-        customAppVariables[key] = value;
+        const customAppVariables = { ..._ref.state.currentState };
```

Comment:
Let's fix this variable naming to `currentState` instead of `customAppVariables` as it is pointing to the current state.